### PR TITLE
Ads api

### PR DESCRIPTION
**Pull request by diogoaihara**
Migration from Adwords API to Ads API

### CODE REVIEW

File: megalist_dataflow/uploaders/google_ads/customer_match/abstract_uploader.py [hunk 1]
```diff
@@ -39,51 +39,80 @@ def __init__(self, oauth_credentials: OAuthCredentials, developer_token: str):
...
-      logging.getLogger(_DEFAULT_LOGGER).info('List %s created with id: %d',
-                                              list_name, list_id)
+        '%s list does not exist, creating...', list_name)
+      user_list_service = self._get_user_list_service(customer_id)
+      request = {
+        'customer_id': customer_id,
+        'partial_failure': False,
+        'validate_only': False,
+        'operations': [{
+          # 'create': **list_definition
```
Comment (astivi) [inline megalist_dataflow/uploaders/google_ads/customer_match/abstract_uploader.py:72]:
```suggestion
```

Comment (astivi) [reply megalist_dataflow/uploaders/google_ads/customer_match/abstract_uploader.py:72]: Remove this line.

**Review by astivi — APPROVED** [review]

File: megalist_dataflow/uploaders/google_ads/customer_match/contact_info_uploader.py [hunk 2]
```diff
@@ -26,19 +26,18 @@ class GoogleAdsCustomerMatchContactInfoUploaderDoFn(GoogleAdsCustomerMatchAbstra
...
-        # CRM-based user lists can use a membershipLifeSpan of 10000 to indicate
-        # unlimited; otherwise normal values apply.
-        'membershipLifeSpan': 10000,
-        'uploadKeyType': 'CONTACT_INFO'
+      'membership_status': 'OPEN',
+      'name': list_name,
+      'description': 'List created automatically by Megalista',
+      'membership_life_span': 10000,
+      'crm_based_user_list': {
+        'upload_key_type': 'CONTACT_INFO', #CONTACT_INFO, CRM_ID, MOBILE_ADVERTISING_ID
```
Comment (antoniolmm) [inline megalist_dataflow/uploaders/google_ads/customer_match/contact_info_uploader.py:34]:
```suggestion
        'upload_key_type': 'CONTACT_INFO',
```

File: megalist_dataflow/uploaders/google_ads/customer_match/mobile_uploader.py [hunk 3]
```diff
@@ -33,20 +33,19 @@ def get_list_definition(self, account_config: AccountConfig, destination_metadat
...
-        # unlimited; otherwise normal values apply.
-        'membershipLifeSpan': 10000,
-        'appId': app_id,
-        'uploadKeyType': 'MOBILE_ADVERTISING_ID'
+      'membership_status': 'OPEN',
+      'name': list_name,
+      'description': 'List created automatically by Megalista',
+      'membership_life_span': 10000,
+      'crm_based_user_list': {
+        'upload_key_type': 'MOBILE_ADVERTISING_ID', #CONTACT_INFO, CRM_ID, MOBILE_ADVERTISING_ID
```
Comment (antoniolmm) [inline megalist_dataflow/uploaders/google_ads/customer_match/mobile_uploader.py:41]:
```suggestion
        'upload_key_type': 'MOBILE_ADVERTISING_ID',
```

**Review by antoniolmm — APPROVED** [review]

**Pull request merged**